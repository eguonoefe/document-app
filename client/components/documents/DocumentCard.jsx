import React from 'react';
import PropTypes from 'prop-types';
import parseDate from '../../utilities/parseDate';

/**
* Document Card for viewing a single document
*
* @param {Object} props { title, id, access, updatedAt, onClick, User }
* @returns {String} HTML markup for the Document Card
*/
const DocumentCard = (props) => {
  const { title, id, access, updatedAt, onClick, User } = props;
  return (
    <div>
      <a
        onClick={onClick}
        name={id}
        className="truncate"
        href="#!">{title}</a>
      <p>By {User.firstName} {User.lastName}</p>
      {(access === 'public') && <p>Public</p>}
      {(access === 'private') && <p>Private</p>}
      {(access === 'role') && <p>Role</p>}
      <p>Last Edited: {parseDate(updatedAt)}</p>
      <div className="divider" />
    </div>
  );
};

// Set Props
DocumentCard.propTypes = {
  title: PropTypes.string,
  access: PropTypes.string,
  id: PropTypes.number,
  User: PropTypes.object,
  onClick: PropTypes.func,
  updatedAt: PropTypes.node
};

export default DocumentCard;
