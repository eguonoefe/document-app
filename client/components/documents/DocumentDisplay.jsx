import React from 'react';
import PropTypes from 'prop-types';
import TinyMCE from 'react-tinymce';

import InputField from '../common/InputField';
import Dropdown from '../common/Dropdown';
/**
 * Document Display component for creating a document
 *
 * @param {Object} props { onChange, onSubmit, document, getContent }
 * @returns {String} HTML markup for the Document Display
 */
const DocumentDisplay = ({ onChange, onSubmit, document, getContent }) => (
  <div className="document-form">
    <div className="center hover">
      <h5>Create Document</h5>
      <InputField
        name="title"
        value={document.title}
        placeholder="Document Title"
        className="validate div-design"
        type="text" onChange={onChange} />
      <select
        name="access"
        className="browser-default input-field select" onChange={onChange}>
        <Dropdown value="" text="Select access type" />
        <Dropdown value="public" text="Public" />
        <Dropdown value="private" text="Private" />
        <Dropdown value="role" text="Role" />
      </select>
      <TinyMCE
        content={document.content}
        config={{
          plugins: 'link image code',
          height: 200,
          toolbar:
          'undo redo | bold italic | alignleft aligncenter alignright | code'
        }}
        onChange={getContent}
      />
      <div className="input-field center">
        <button
          className="waves-effect btn button-design" onClick={onSubmit}
          id="save-document">
          Create
          </button>
      </div>
    </div>
  </div>
);

DocumentDisplay.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  document: PropTypes.object.isRequired,
  getContent: PropTypes.func.isRequired
};

export default DocumentDisplay;
