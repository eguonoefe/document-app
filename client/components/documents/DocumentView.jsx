import React from 'react';
import { Modal } from 'react-materialize';
import ReactHtmlParser from 'react-html-parser';
import TinyMCE from 'react-tinymce';
import PropTypes from 'prop-types';

import InputField from '../common/InputField';
import Dropdown from '../common/Dropdown';

/**
 * Document view component for viewing a document
 * Has update and delete modals for deleting and updating a document
 *
 * @param {Object} props { id, title, content, access, onSubmit,
 * onChange, getContent, deleteDocument, userId, authorId }
 * @returns {String} HTML markup for the Document view
 */
const DocumentView = (props) => {
  const {
    id,
    title,
    content,
    access,
    onSubmit,
    onChange,
    getContent,
    deleteDocument,
    userId,
    authorId } = props;
  return (
    <div className="card blue-grey darken-1">
      <div className="card-content white-text">
        <span className="card-title">{title}</span>
        <div>{ReactHtmlParser(content)}</div>
        <p>{access}</p>
      </div>
      <div className="card-action">
        <div className="row">
          <div className="col l6 m6 s12">
            {userId === authorId && <Modal
              trigger={
                <a
                  className="waves-effect btn button-design"
                  data-target="passwordModal" id="delete-document">
                  Delete
                </a>}
              actions={
                <div>
                  <button
                    className="btn btn-flat modal-action modal-close"
                    name={id} onClick={deleteDocument}
                    id="delete">
                    Delete
                </button>
                  <button
                    className="btn btn-flat modal-action modal-close left">
                    Cancel
                  </button>
                </div>}>
              <div >
                <h5
                  className="center">
                  Are you sure you want to delete the user
                </h5>
              </div>
            </Modal>
            }
          </div>
          <div className="col l6 m6 s12">
            {userId === authorId && <Modal
              trigger={
                <a
                  className="waves-effect btn button-design"
                  data-target="passwordModal"
                  id="update-document"
                >
                  Update
                </a>
              }
              actions={
                <div>
                  <button
                    id="save-update"
                    className="modal-action modal-close btn btn-flat"
                    onClick={onSubmit}>
                    Update
                      </button>
                  <button
                    className="btn btn-flat modal-action modal-close left">
                    Cancel
                      </button>
                </div>
              }>
              <div >
                <h5>Update Document</h5>
                <InputField
                  name="title"
                  value={title}
                  placeholder="Document Title"
                  className="validate form-design"
                  type="text" onChange={onChange} />
                <TinyMCE
                  content={content}
                  config={{
                    plugins: 'link image code',
                    height: 200,
                    toolbar:
                    `undo redo |
                    bold italic | alignleft aligncenter alignright | code`
                  }}
                  onChange={getContent}
                />
                <select
                  name="access"
                  value={access}
                  className="browser-default input-field select"
                  onChange={onChange}>
                  <Dropdown value="" text="Select access type" />
                  <Dropdown value="public" text="Public" />
                  <Dropdown value="private" text="Private" />
                  <Dropdown value="role" text="Role" />
                </select>
              </div>
            </Modal>}
          </div>
        </div>
      </div>
    </div>
  );
};

DocumentView.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  getContent: PropTypes.func,
  deleteDocument: PropTypes.func.isRequired,
  id: PropTypes.number,
  userId: PropTypes.number,
  authorId: PropTypes.number,
  title: PropTypes.string,
  content: PropTypes.string,
  access: PropTypes.string
};

export default DocumentView;
