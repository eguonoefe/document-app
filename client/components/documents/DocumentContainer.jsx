import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import DocumentDisplay from './DocumentDisplay';
import DocumentActions from '../../actions/DocumentActions';
import validate from '../../utilities/validate';

const createDocument = DocumentActions.createDocument;

/**
 * @class DocumentContainer
 * @extends {Component}
 */
export class DocumentContainer extends Component {
  /**
   * Creates an instance of DocumentContainer.
   * Binds the functions to the class
   * @param {Object} props -
   * @memberof DocumentContainer
   */
  constructor(props) {
    super(props);
    this.state = {
      title: '',
      content: '',
      access: ''
    };
    this.onSubmit = this.onSubmit.bind(this);
    this.onChange = this.onChange.bind(this);
    this.getContent = this.getContent.bind(this);
  }

  /**
  * Validates the DocumentDisplay input fields
  * Makes an action call to create a new document
  * Toasts the error/success message
  * @return {void}
  * @memberof DocumentContainer
  */
  onSubmit() {
    try {
      const { valid } = validate.validateSaveDocument(this.state);
      if (!valid) {
        throw new Error('No field should be left blank');
      }
      this.props.createDocument(this.state)
        .then(() => {
          if (this.props.document.message) {
            return Materialize.toast(
              this.props.document.message, 2000,
              'indigo darken-4 white-text rounded');
          }
          Materialize.toast(
            'Document Created', 2000, 'indigo darken-4 white-text rounded');
          this.props.history.push('/dashboard');
        });
    } catch (err) {
      Materialize.toast(err.message, 3000,
        'indigo darken-4 white-text rounded');
    }
  }
  /**
  * Sets the event value to the state
  * @return {void}
  * @param {Object} event The event of the HTML component
  * @memberof DocumentContainer
  */
  onChange(event) {
    const field = event.target.name;
    this.setState({ [field]: event.target.value });
  }

  /**
  * Get the content of the TinyMCE editor and sets it to the state
  *
  * @param {Object} event -
  * @returns {void}
  */
  getContent(event) {
    this.setState({ content: event.target.getContent() });
  }
  /**
  * Renders the DocumentContainer component
  * @returns {String} The HTML markup for the DocumentContainer
  * @memberof DocumentContainer
  */
  render() {
    return (
      <DocumentDisplay
        onSubmit={this.onSubmit}
        onChange={this.onChange}
        document={this.state}
        getContent={this.getContent} />
    );
  }
}

const mapPropsToState = state => (
  {
    document: state.document
  }
);

DocumentContainer.propTypes = {
  createDocument: PropTypes.func.isRequired,
  document: PropTypes.object.isRequired,
  history: PropTypes.object
};

export default connect(
  mapPropsToState, { createDocument })(withRouter(DocumentContainer));
