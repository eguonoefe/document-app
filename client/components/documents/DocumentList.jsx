import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Switch, Route, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';

import DocumentActions from '../../actions/DocumentActions';
import DocumentCard from './DocumentCard';
import DocumentView from './DocumentView';
import Dropdown from '../common/Dropdown';

const getAllDocuments = DocumentActions.getAllDocuments;
const viewDocument = DocumentActions.viewDocument;
const updateDocument = DocumentActions.updateDocument;
const deleteDocument = DocumentActions.deleteDocument;
const searchDocuments = DocumentActions.searchDocuments;
const getUserDocuments = DocumentActions.getUserDocuments;

/**
 * @class DocumentList
 * @extends {Component}
 */
export class DocumentList extends Component {
  /**
   * Creates an instance of DocumentList.
   * Binds class methods
   * @param {Object} props -
   * @memberof DocumentList
   */
  constructor(props) {
    super(props);
    this.state = {
      documents: [],
      document: {},
      offset: 0,
      pageCount: 0,
      searchTerm: '',
      docquery: '',
      personal: false
    };
    this.onClick = this.onClick.bind(this);
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.deleteDocument = this.deleteDocument.bind(this);
    this.onSearch = this.onSearch.bind(this);
    this.changeDocument = this.changeDocument.bind(this);
    this.handlePageClick = this.handlePageClick.bind(this);
    this.getContent = this.getContent.bind(this);
  }

  /**
   * Makes action call to retrieve list of documents
   *
   * @return {void}
   * @memberof DocumentList
   */
  componentDidMount() {
    this.updateDocumentList();
  }
  /**
   * Update the state if the props are changed
   *
   * @return {void}
   * @param {Object} nextProps -
   * @memberof DocumentList
   */
  componentWillReceiveProps(nextProps) {
    if (this.props !== nextProps) {
      this.setState({
        documents: nextProps.documentList,
        pageCount: nextProps.pagination.pageCount,
        count: nextProps.pagination.count
      });
    }
  }
  /**
  * Makes an action call to get personal/all documents
  * Sets document list to the state
  *
  * @return {void}
  * @param {Object} event -
  * @memberof DocumentList
  */
  changeDocument(event) {
    const value = event.target.value;
    this.setState({
      [event.target.name]: value
    });
    if (value === 'Personal') {
      this.props.getUserDocuments(this.props.access.user.id)
        .then(() => {
          this.setState({
            documents: this.props.documentList,
            personal: true
          });
        });
    }
    if (value === 'All') {
      this.updateDocumentList();
      this.setState({
        personal: false
      });
    }
  }
  /**
  * Sets the event value to the state
  * @return {void}
  * @param {Object} event The event of the HTML component
  * @memberof DocumentContainer
  */
  onChange(event) {
    const field = event.target.name;
    this.setState({ [field]: event.target.value });
  }
  /**
  * Makes an action call to view a document
  * Toasts the error/success message
  * Sets document to the state
  *
  * @param {Object} event -
  * @return {void}
  * @memberof DocumentList
  */
  onClick(event) {
    this.props.viewDocument(event.target.name).then(() => {
      if (this.props.document.message) {
        return Materialize.toast(
          this.props.document.message, 2000,
          'indigo darken-4 white-text rounded');
      }
      this.setState({
        title: this.props.document.title,
        content: this.props.document.content,
        access: this.props.document.access
      });
      this.props.history.push(`${this.props.match.url}/view-document`);
    });
  }

  /**
   * Sets the searchterm to the state
   * Makes an action call to the search for a document
   * @param {Object} event -
   * @return {void}
   * @memberof DocumentList
   */
  onSearch(event) {
    this.setState({
      searchTerm: event.target.value
    });
    this.props.searchDocuments(event.target.value)
      .then(() => {
      });
  }

  /**
   * Makes an action call to get all documents
   * Sets list of documents to the state
   *
   * @return {void}
   * @memberof DocumentList
   */
  updateDocumentList() {
    this.props.getAllDocuments()
      .then(() => {
        this.setState({
          documents: this.props.documentList,
          count: this.props.pagination.count
        });
      });
  }
  /**
  * Get the content of the TinyMCE editor and sets it to the state
  *
  * @param {Object} event -
  * @returns {void} nothing
  */
  getContent(event) {
    this.setState({ content: event.target.getContent() });
  }
  /**
  * Makes an action call to delete a document
  * Toasts the error/success message
  *
  * @return {void}
  * @param {Object} event -
  * @memberof DocumentList
  */
  deleteDocument(event) {
    this.props.deleteDocument(event.target.name).then(() => {
      Materialize.toast(
        this.props.document.message, 2000,
        'indigo darken-4 white-text rounded');
      this.props.history.push(`${this.props.match.url}`);
    });
  }

  /**
  * Pagination for the list of documents
  *
  * @return {void}
  * @param {Object} data -
  * @memberof DocumentList
  */
  handlePageClick(data) {
    const selected = data.selected;
    const limit = 5;
    const offset = Math.ceil(selected * limit);
    this.setState({ offset });
    if (this.state.personal) {
      this.props.getUserDocuments(
        this.props.access.user.id, offset, limit).then(() => {
          this.setState({
            documents: this.props.documentList,
            pageCount: this.props.pagination.pageCount,
            count: this.props.pagination.count
          });
        });
    } else {
      this.props.getAllDocuments(offset, limit).then(() => {
        this.setState({
          documents: this.props.documentList,
          pageCount: this.props.pagination.pageCount,
          count: this.props.pagination.count
        });
      });
    }
  }
  /**
  * Makes an action call to update a document
  * Toasts error/success message
  * Sets updated document to state
  *
  * @return {void}
  * @memberof DocumentList
  */
  onSubmit() {
    const updatedDocument = {
      title: this.state.title,
      content: this.state.content,
      access: this.state.access
    };
    this.props.updateDocument(this.props.document.id, updatedDocument)
      .then(() => {
        if (this.props.message) {
          return Materialize.toast(
            this.props.message, 2000,
            'indigo darken-4 white-text rounded');
        }
        Materialize.toast(
          'Document has been updated',
          2000, 'indigo darken-4 white-text rounded'
        );
        this.props.history.push('/dashboard');
        this.setState({
          title: this.props.document.title,
          content: this.props.document.content,
          access: this.props.document.access
        });
      });
  }

  /**
  * Renders the Document List component
  *
  * @returns {String} HTML markup for the DocumentList
  * @memberof DocumentList
  */
  render() {
    return (
      <div className="document-list">
        <div className="container">
          <div className="row">
            <div className="col l12 m12 s12">
              <div className="col l5 m5 s12">
                <input
                  className="search"
                  type="text"
                  name="searchTerm"
                  id="documentSearch"
                  placeholder="Search.."
                  onChange={this.onSearch} />
              </div>
              <div className="col l6 m6 s12 center">
                <select
                  name="docquery"
                  className="browser-default input-field select"
                  onChange={this.changeDocument}>
                  <Dropdown value="All" text="All Documents" />
                  <Dropdown value="Personal" text="Personal" />
                </select>
              </div>
            </div>
            <div className="document-list-view">
              <div className="col l5 m5 s12">
                <div className=" card-panel hoverable">
                  {(this.state.docquery === 'All' || this.state.docquery === '')
                    && <h5>All Documents</h5>}
                  {(this.state.docquery === 'Personal')
                    && <h5>Personal Documents</h5>}
                  <div className="scrollable">
                    {this.state.documents.map(document => (
                      <DocumentCard
                        key={document.id} {...document}
                        onClick={this.onClick} match={this.props.match}
                      />
                    ))}
                    {(this.state.count > 5) && <ReactPaginate
                      previousLabel={'previous'}
                      nextLabel={'next'}
                      pageCount={this.state.pageCount}
                      marginPagesDisplayed={1}
                      pageRangeDisplayed={3}
                      onPageChange={this.handlePageClick}
                      containerClassName={'pagination'}
                      subContainerClassName={'pages pagination'}
                      activeClassName={'active'}
                    />}
                  </div>
                </div>
              </div>
            </div>
            <div className="col l6 m6 s12 text-scrollable">
              <Switch>
                <Route
                  // eslint-disable-next-line
                  path={`${this.props.match.url}/view-document`} render={() => {
                    if (!this.props.document.id) {
                      this.props.history.push(`${this.props.match.url}`);
                    }
                    return (
                      <DocumentView
                        id={this.props.document.id}
                        authorId={this.props.document.authorId}
                        title={this.state.title}
                        content={this.state.content}
                        access={this.state.access}
                        onChange={this.onChange}
                        onSubmit={this.onSubmit}
                        getContent={this.getContent}
                        deleteDocument={this.deleteDocument}
                        userId={this.props.access.user.id} />);
                  }} />
              </Switch>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

DocumentList.propTypes = {
  access: PropTypes.object,
  document: PropTypes.object,
  documentList: PropTypes.array,
  getAllDocuments: PropTypes.func,
  getUserDocuments: PropTypes.func,
  viewDocument: PropTypes.func,
  match: PropTypes.object,
  searchDocuments: PropTypes.func,
  deleteDocument: PropTypes.func,
  updateDocument: PropTypes.func,
  pagination: PropTypes.object,
  message: PropTypes.string,
  history: PropTypes.object
};

const mapPropsToState = state => (
  {
    documentList: state.document.documentList,
    document: state.document.document,
    message: state.document.message,
    pagination: state.document.pagination,
    access: state.access
  }
);

export default connect(
  mapPropsToState, {
    getAllDocuments,
    viewDocument,
    updateDocument,
    deleteDocument,
    searchDocuments,
    getUserDocuments
  })(withRouter(DocumentList));
