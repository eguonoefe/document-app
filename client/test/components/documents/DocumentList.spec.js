/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { DocumentList } from '../../../components/documents/DocumentList';

configure({ adapter: new Adapter() });

describe('Document List', () => {
  const props = {
    access: {
      isAuthenticated: true,
      user: {
        id: 1, email: 'efeeguono@gmail.com'
      }
    },
    documentList: [{ title: 'Manny' },
    { title: 'John' }, { title: 'move' }],
    document: { title: 'Eguono', id: 3, content: 'john', access: 'role' },
    getAllDocuments: jest.fn(() => {
      return Promise.resolve();
    }),
    viewDocument: jest.fn(() => {
      return Promise.resolve();
    }),
    updateDocument: jest.fn(() => {
      return Promise.resolve();
    }),
    searchDocuments: jest.fn(() => {
      return Promise.resolve();
    }),
    getUserDocuments: jest.fn(() => {
      return Promise.resolve();
    }),
    deleteDocument: jest.fn(() => {
      return Promise.resolve();
    }),
    history: {
      push: jest.fn()
    },
    match: {
      url: ''
    },
  };
  const component = shallow(<DocumentList {...props} />);
  beforeEach(() => {
    global.Materialize = { toast: jest.fn() };
  });
  describe('Container', () => {
    test('should match snapshot of document list', () => {
      const tree = toJson(component);
      expect(tree).toMatchSnapshot();
    });
    test('onSearch function should run when called', () => {
      const spy = jest.spyOn(component.instance(), 'componentDidMount');
      component.instance().componentDidMount();
      expect(spy).toHaveBeenCalled();
      component.instance().onSearch(
        { target: { value: 'Manny', name: 'searchTerm' } });
      expect(component.state('searchTerm')).toEqual('Manny');
    });
    test('onChange function should run when called', () => {
      component.instance().onChange(
        { target: { value: 'eguono', name: 'title' } });
      component.instance().onChange(
        { target: { value: 'efe', name: 'content' } });
      component.instance().onChange(
        { target: { value: 'public', name: 'access' } });
      expect(component.state('title')).toEqual('eguono');
      expect(component.state('content')).toEqual('efe');
      expect(component.state('access')).toEqual('public');
      const spy = jest.spyOn(component.instance(), 'onSubmit');
      component.instance().onSubmit();
      expect(spy).toHaveBeenCalled();
    });
    test('deleteDocument function should run when called', () => {
      const spy = jest.spyOn(component.instance(), 'deleteDocument');
      component.instance().deleteDocument(
        { target: { value: 3, name: 'title' } });
      expect(spy).toHaveBeenCalled();
    });
    test('componentWillRecieveProps will run if new props are added', () => {
      const spy = jest.spyOn(component.instance(), 'componentWillReceiveProps');
      const documentList = [{ title: 'john' }];
      const pagination = { pageCount: '' };
      component.setProps({ documentList, pagination });
      expect(spy).toHaveBeenCalled();
    });
    test('changeDocument function should run when called', () => {
      const spy = jest.spyOn(component.instance(), 'changeDocument');
      const select = component.find('select');
      expect(select.length).toEqual(1);
      component.instance().changeDocument({
        target: { value: 'Personal', name: 'docquery' }
      });
      expect(component.state('docquery')).toEqual('Personal');
      expect(spy).toHaveBeenCalled();
      component.instance().changeDocument({
        target: { value: 'All', name: 'docquery' }
      });
      expect(component.state('docquery')).toEqual('All');
    });
    test('onClick function should run when called', () => {
      const spy = jest.spyOn(component.instance(), 'onClick');
      component.instance().onClick({
        target: { name: 1 }
      });
      expect(spy).toHaveBeenCalled();
    });
  });
});
