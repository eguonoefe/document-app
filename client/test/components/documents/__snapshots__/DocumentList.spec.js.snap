// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document List Container should match snapshot of document list 1`] = `
<div
  className="document-list"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col l12 m12 s12"
      >
        <div
          className="col l5 m5 s12"
        >
          <input
            className="search"
            id="documentSearch"
            name="searchTerm"
            onChange={[Function]}
            placeholder="Search.."
            type="text"
          />
        </div>
        <div
          className="col l6 m6 s12 center"
        >
          <select
            className="browser-default input-field select"
            name="docquery"
            onChange={[Function]}
          >
            <Dropdown
              text="All Documents"
              value="All"
            />
            <Dropdown
              text="Personal"
              value="Personal"
            />
          </select>
        </div>
      </div>
      <div
        className="document-list-view"
      >
        <div
          className="col l5 m5 s12"
        >
          <div
            className=" card-panel hoverable"
          >
            <h5>
              All Documents
            </h5>
            <div
              className="scrollable"
            />
          </div>
        </div>
      </div>
      <div
        className="col l6 m6 s12 text-scrollable"
      >
        <Switch>
          <Route
            path="/view-document"
            render={[Function]}
          />
        </Switch>
      </div>
    </div>
  </div>
</div>
`;
